packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -683,12 +709,62 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		stoppedAt: true,
 	};
 
+	private annotationFields = {
+		id: true,
+		vote: true,
+	};
+
 	async findManyByRangeQuery(query: ExecutionSummaries.RangeQuery): Promise<ExecutionSummary[]> {
 		if (query?.accessibleWorkflowIds?.length === 0) {
 			throw new ApplicationError('Expected accessible workflow IDs');
 		}
 
-		const executions: ExecutionSummary[] = await this.toQueryBuilder(query).getRawMany();
+		const qb = this.toQueryBuilderWithAnnotations(query);
+
+		// FIXME: This needs refactoring
+		const rawExecutionsWithTags: Array<
+			ExecutionSummary & {
+				'annotation.id': number;
+				'annotation.vote': AnnotationVote;
+				'annotation.tags.id': string;
+				'annotation.tags.name': string;
+			}
+		> = await qb.getRawMany();
+
+		const executions = rawExecutionsWithTags.reduce(
+			(
+				acc,
+				{
+					'annotation.id': _,
+					'annotation.vote': vote,
+					'annotation.tags.id': tagId,
+					'annotation.tags.name': tagName,
+					...row
+				},
+			) => {
+				const existingExecution = acc.find((e) => e.id === row.id);
+
+				if (existingExecution) {
+					if (tagId) {
+						existingExecution.annotation = existingExecution.annotation ?? {
+							vote,
+							tags: [] as Array<{ id: string; name: string }>,
+						};
+						existingExecution.annotation.tags.push({ id: tagId, name: tagName });
+					}
+				} else {
+					acc.push({
+						...row,
+						annotation: {
+							vote,
+							tags: tagId ? [{ id: tagId, name: tagName }] : [],
+						},
+					});
+				}
+				return acc;
+			},
+			[] as ExecutionSummary[],
+		);
```

Comment:
Do you have ideas on how to avoid manual reduce at all? I have a feeling like I'm reinventing a wheel here. This is something typeorm does under the hood, when querying the data with relations via standard repository methods, but I haven't figured out how to make it nice with query builder.

Reply:
Unfortunately I'm not aware that typeorm exposes how it mangles rows into entities :/ I guess that's the price we have to pay if we use raw queries instead the built in ones. Other would be to query the tags separately from the executions and combine them afterwards. That would be 2 queries instead of one, but the code would probably be simpler

Reply:
I've made some refactoring, hope this is a bit easier to read now